biogeochem/EDPhysiologyMod.F90

```diff
@@ -1189,223 +1575,345 @@ subroutine CWD_Input( currentSite, currentPatch)
+
+      plant_dens =  currentCohort%n/currentPatch%area
+
+      ! ---------------------------------------------------------------------------------
+      ! PART 1 Litter fluxes from non-mortal tissue turnovers  Kg/m2/day
+      !        Important note:  Turnover has already been removed from the cohorts.
+      !        So, in the next part of this algorithm, when we send the biomass
+      !        from dying trees to the litter pools, we don't have to worry 
+      !        about double counting.
+      ! ---------------------------------------------------------------------------------
+
+      flux_diags%leaf_litter_input(pft) = &
+            flux_diags%leaf_litter_input(pft) +  &
+            leaf_m_turnover * currentCohort%n
+      
+      root_fines_tot = (fnrt_m_turnover + store_m_turnover ) * &
+            plant_dens
 
-      leaf_c          = currentCohort%prt%GetState(leaf_organ,all_carbon_elements)
-      store_c         = currentCohort%prt%GetState(store_organ,all_carbon_elements)
-      fnrt_c          = currentCohort%prt%GetState(fnrt_organ,all_carbon_elements)
-      sapw_c          = currentCohort%prt%GetState(sapw_organ,all_carbon_elements)
-      struct_c        = currentCohort%prt%GetState(struct_organ,all_carbon_elements)
+      do dcmpy=1,ndcmpy
+          dcmpy_frac = GetDecompyFrac(pft,dcmpy)
 
-      ! ================================================        
-      ! Litter from tissue turnover. KgC/m2/year
-      ! ================================================   
+          litt%leaf_fines_in(dcmpy) = litt%leaf_fines_in(dcmpy) + &
+                leaf_m_turnover * plant_dens * dcmpy_frac
 
-      currentPatch%leaf_litter_in(pft) = currentPatch%leaf_litter_in(pft) + &
-           leaf_c_turnover * currentCohort%n/currentPatch%area/hlm_freq_day
+          do ilyr = 1, numlevsoil
+              litt%root_fines_in(dcmpy,ilyr) = litt%root_fines_in(dcmpy,ilyr) + &
+                    currentSite%rootfrac_scr(ilyr) * root_fines_tot * dcmpy_frac
+          end do
+      end do
       
-      currentPatch%root_litter_in(pft) = currentPatch%root_litter_in(pft) + &
-           (fnrt_c_turnover + store_c_turnover ) * &
-           currentCohort%n/currentPatch%area/hlm_freq_day
+      flux_diags%root_litter_input(pft) = &
+            flux_diags%root_litter_input(pft) +  &
+            (fnrt_m_turnover + store_m_turnover ) * currentCohort%n
       
       
-      !daily leaf loss needs to be scaled up to the annual scale here. 
-
-      ! ---------------------------------------------------------------------------------
       ! Assumption: turnover from deadwood and sapwood are lumped together in CWD pool
-      ! ---------------------------------------------------------------------------------
       
       do c = 1,ncwd
-         currentPatch%cwd_AG_in(c) = currentPatch%cwd_AG_in(c) + &
-              (sapw_c_turnover + struct_c_turnover)/hlm_freq_day   * &
```

Comment:
Its due to a change in the units of the variable.  Previously, litter fluxes had units of per year, now they are represented in units of kg/m2/day.   This is more consistent with their use, because the flux is actually applied daily.

Reply:
Nice.